src/components/MoneyReportHeader.tsx

```diff
@@ -207,7 +208,13 @@ function MoneyReportHeader({policy, report: moneyRequestReport, transactionThrea
     const isFromPaidPolicy = policyType === CONST.POLICY.TYPE.TEAM || policyType === CONST.POLICY.TYPE.CORPORATE;
     const shouldShowStatusBar =
         hasAllPendingRTERViolations || shouldShowBrokenConnectionViolation || hasOnlyHeldExpenses || hasScanningReceipt || isPayAtEndExpense || hasOnlyPendingTransactions;
-    const shouldShowNextStep = transactions?.length !== 0 && isFromPaidPolicy && !!nextStep?.message?.length && !shouldShowStatusBar;
+
+    // When prevent self-approval is enabled, we need to show the optimistic next step
+    // We should always show this optimistic message for policies with preventSelfApproval
+    // to avoid any flicker during transitions between online/offline states
+    const optimisticNextStep = policy?.preventSelfApproval ? buildOptimisticNextStepForPreventSelfApprovalsEnabled() : nextStep;
+
+    const shouldShowNextStep = transactions?.length !== 0 && isFromPaidPolicy && !!optimisticNextStep?.message?.length && !shouldShowStatusBar;
```

Comment:
Hmm shouldn't we only be doing `buildOptimisticNextStepForPreventSelfApprovalsEnabled` if `policy?.preventSelfApproval` AND if the submitter is the next receiver?

Reply:
More investigation in https://expensify.slack.com/archives/C06ML6X0W9L/p1741645160823569 - but yeah OldDot is correctly showing the next steps as "Waiting for <manager> to approve expenses" b/c I'm not submitting to myself 😬 I thinkkkkk i can put up a quick fix

Reply:
created https://github.com/Expensify/App/issues/58128 to fix